src/components/TrafficList/TrafficListComponent.tsx

```diff
@@ -0,0 +1,353 @@
+import * as React from 'react';
+import { Badge, Title, Tooltip, TooltipPosition } from '@patternfly/react-core';
+import { style } from 'typestyle';
+import {
+  IRow,
+  ISortBy,
+  sortable,
+  SortByDirection,
+  Table,
+  TableBody,
+  TableHeader,
+  cellWidth
+} from '@patternfly/react-table';
+import { Link } from 'react-router-dom';
+import { TrafficItem, TrafficNode, TrafficDirection } from './TrafficDetails';
+import * as FilterComponent from '../FilterList/FilterComponent';
+import { ThresholdStatus, NA } from 'types/Health';
+import { NodeType, hasProtocolTraffic, ProtocolTraffic } from 'types/Graph';
+import { getTrafficHealth } from 'types/ErrorRate';
+import history, { URLParam } from 'app/History';
+import { createIcon } from 'components/Health/Helper';
+import { sortFields } from './FiltersAndSorts';
+import { SortField } from 'types/SortFilters';
+
+export interface TrafficListItem {
+  direction: TrafficDirection;
+  healthStatus: ThresholdStatus;
+  icon: string;
+  node: TrafficNode;
+  protocol: string;
+  trafficRate: string;
+  trafficPercent: string; // percent success
+}
+
+type TrafficListComponentProps = FilterComponent.Props<TrafficListItem> & {
+  trafficItems: TrafficItem[];
+};
+
+type TrafficListComponentState = FilterComponent.State<TrafficListItem> & {
+  sortBy: ISortBy;
```

Comment:
I'm wondering whether it's hard to remove these two apparently redundant state fields ?
I know this component tends to not change a lot, but in some form I see these fields need to be in sync with the inherited state...

I'm thinking this could be a source of mistakes when modifying the code and forgetting to sync the state. But if it's hard, let it be.

Reply:
@israel-hdez I will look, it's possible it's an unnecessary holdover from the example impl I used originally.

Reply:
@israel-hdez Yes, I was able to remove the redundant sort state given the other refactoring already done in the PR.